extensions/redis-cache/runtime/src/main/java/io/quarkus/cache/redis/runtime/RedisCacheImpl.java

```diff
@@ -218,6 +218,25 @@ public Uni<V> apply(RedisConnection connection) {
         });
     }
 
+    @Override
+    public <K, V> V execute(K key, Class<V> clazz, BiFunction<K, V, V> command) {
+        byte[] encodedKey = marshaller.encode(computeActualKey(encodeKey(key)));
+        return withConnection(connection ->
+                watch(connection, encodedKey)
+                        .chain(new GetFromConnectionSupplier<>(connection, clazz, encodedKey, marshaller))
+                        .chain(previousValue ->
+                                Uni.createFrom().item(() -> command.apply(key, previousValue))
+                                        .runSubscriptionOn(MutinyHelper.blockingExecutor(vertx.getDelegate()))
+                        ).chain(value -> {
+                            if (value == null) {
+                                throw new IllegalArgumentException("Cannot cache `null` value");
+                            }
+                            byte[] encodedValue = marshaller.encode(value);
+                            return multi(connection, set(connection, encodedKey, encodedValue)).replaceWith(value);
+                        })
+        ).await().indefinitely();
```

Comment:
Don't use indefinitely(), probably better to use the default timeout from redis.

Reply:
I added a runtime config item per cache, is this ok ?